### PR TITLE
Update to the jupyter walkthrough

### PR DESCRIPTION
**Pull request by aronchick**

### CODE REVIEW

File: JupyterNotebookFlow.md [hunk 1]
```diff
@@ -25,7 +31,10 @@ Before you follow along, please make sure that you have ``Python3``, ``Java JRE`
       random_forest.py
       wine-quality.csv
    ```
-
+**Questions about this step**
+- How were these things created? Auto generated?
+- Could we start with ccds to create the repo?
```
Comment (luzhang06) [inline JupyterNotebookFlow.md:36]: Do data scientists start with creating a repo first or they start from some notebook/python files.

File: JupyterNotebookFlow.md [hunk 2]
```diff
@@ -25,7 +31,10 @@ Before you follow along, please make sure that you have ``Python3``, ``Java JRE`
       random_forest.py
       wine-quality.csv
    ```
-
+**Questions about this step**
+- How were these things created? Auto generated?
+- Could we start with ccds to create the repo? 
+- Where'd she get the csv? Starting with data is a really big issue, we should take it on.
```
Comment (luzhang06) [inline JupyterNotebookFlow.md:37]: Data exploration might take most of data scientists' time at early stage. We should consider this flow.

File: JupyterNotebookFlow.md (hunk 1 above)
Comment (aronchick) [reply JupyterNotebookFlow.md:36]: I think that's the challenge - they PROBABLY start locally, but the experience of then creating a repo and adding your files is not elegant.

File: JupyterNotebookFlow.md (hunk 2 above)
Comment (aronchick) [reply JupyterNotebookFlow.md:37]: Completely agree, was talking with folks earlier about it. We need to address this straight on.

File: JupyterNotebookFlow.md [hunk 3]
```diff
@@ -36,27 +45,52 @@ Before you follow along, please make sure that you have ``Python3``, ``Java JRE`
...
+**Questions about this step**
+- Is there anything we can do to hightlight issues along the way? Easier debugging/etc?
+- What about auto-logging - reading stuff so the data scientist doesn't have to?
 
 3. The metrics and artifacts are stored in the specific folders under the local project folder ```mlflow/examples/h2o/mlruns/```.
 
    ![image-metrics-artifacts-folder](_assets/JupyterNotebookFlow/image-metrics-artifacts-folder.png)
 
+**Questions about this step**
+- Why are we using h2o?
```
Comment (luzhang06) [inline JupyterNotebookFlow.md:57]: It's just a random choice from the mlflow examples. We can change to other example which is appropriate.

File: JupyterNotebookFlow.md [hunk 4]
```diff
@@ -36,27 +45,52 @@ Before you follow along, please make sure that you have ``Python3``, ``Java JRE`
    ```
    > Note: you can also open ```random_forest.ipynb``` in Jupyter Notebook to execute one step at the time.
 
+**Questions about this step**
+- Is there anything we can do to hightlight issues along the way? Easier debugging/etc?
```
Comment (luzhang06) [inline JupyterNotebookFlow.md:49]: Any thoughts you have in mind for easier debugging experience?

File: JupyterNotebookFlow.md [hunk 5]
```diff
@@ -36,27 +45,52 @@ Before you follow along, please make sure that you have ``Python3``, ``Java JRE`
...
+- What about auto-logging - reading stuff so the data scientist doesn't have to?
```
Comment (luzhang06) [inline JupyterNotebookFlow.md:50]: That would be the ideal experience if we can achieve it. mlflow is just an example to show case. It's not necessarily stick to mlflow current experience.

File: JupyterNotebookFlow.md [hunk 6]
```diff
@@ -12,8 +12,14 @@ However, these issues could be elegantly resolved with Microsoft's renewed visio
```
Comment (aronchick) [inline JupyterNotebookFlow.md:12]: - User does whatever they want locally - when they check in, are prompted to "promote" repo to become an "ML" repo
	- promote could mean lay down empty folders
- steal from repo2docker - GHA triggered, see if MLproj config file is there, detect common artifacts, if present, will generate MLproj artifacts
	- Automatically wire up pushing to octostore (optionally)
	- Once the mlproj config file is generated, you can add ignores/etc
	- GHA may not overwrite mlproj which will prevent re-running accidentally
- Metadata scan - https://opentelemetry.io/docs/collector/about/
	- Doesn't convert anything so they can still use MLFlow locally
	- Merging/unmerging issue
	- Can set (in mlproj) what the merge strategy is
	- Borrow from https://docs.metaflow.org/getting-started/tutorials (https://towardsdatascience.com/learn-metaflow-in-10-mins-netflixs-python-r-framework-for-data-scientists-2ef124c716e4, https://www.youtube.com/watch?v=XV5VGddmP24)
	- https://sgitario.github.io/netflix-machine-learning-infra/

File: JupyterNotebookFlow.md (hunk 1 above)
Comment (aronchick) [reply JupyterNotebookFlow.md:36]:
```
jupyter notebook --create --plugin=dataloader(data=foobar.csv)
nteract <= # https://blog.nteract.io/nteract-revolutionizing-the-notebook-experience-d106ca5d2c38

# Auto creates line below - something that lets me build a dataframe automatically
pd = panda.load('foobar.csv')

pd.visualize() # spits out a useful exploration widget 

pd.first_column # <= once i type 'pd.f...' - 'first_column' is intellisensed

# Could also work against common data stores (mysql, hadoop, mongo) - pre-caches schemas, and top 100 entries

File: JupyterNotebookFlow.md (hunk 4 above)
Comment (aronchick) [reply JupyterNotebookFlow.md:49]: you're in a vscode experience, what do you get
- stop, inspect, continue
- linting
- watch frames
- ...

File: JupyterNotebookFlow.md (hunk 6 above)
Comment (aronchick) [reply JupyterNotebookFlow.md:12]: How do we help people run workflows locally - use the .github/workflow/ml-workflow.yaml

use https://github.com/nektos/act (?)

**Pull request merged**